### PR TITLE
Refactors pathing and featherdoor access

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[CLEANLINESS]
## About the PR 
Aloe pointed out the pathfinding code was wonky, this is the refactored version that also removes the `alien` var I added due to pathing jank.

## Why's this needed? 
f l o c k r e v i e w

### CODE REVIEW

File: code/obj/machinery/door/feather.dm [hunk 1]
```diff
@@ -147,7 +149,9 @@
 ////////////////////
 // everyone allowed to open
 /obj/machinery/door/feather/friendly
```
Comment (FlameArrow57) [inline code/obj/machinery/door/feather.dm:151]: Empty?

Comment (TobleroneSwordfish) [reply code/obj/machinery/door/feather.dm:151]: empty?

Comment (FlameArrow57) [reply code/obj/machinery/door/feather.dm:151]: It's just a proc with no body

Comment (FlameArrow57) [reply code/obj/machinery/door/feather.dm:151]: Oh oops

**Review by frawhst — APPROVED** [review]

**Review by stonepillars — APPROVED** [review]

**Pull request merged**